hayeon/programmers-1/소수-만들기.js

```diff
@@ -0,0 +1,34 @@
+function solution(nums) {
+  let result = 0;
+
+  for (let i = 0; i < nums.length; i++) {
+    for (let j = i + 1; j < nums.length; j++) {
+      for (let k = j + 1; k < nums.length; k++) {
+        const sum = nums[i] + nums[j] + nums[k];
+
+        if (isPrime(sum)) result++;
+      }
+    }
+  }
+  return result;
+}
+
+function isPrime(sum) {
+  for (let i = 2; i < sum; i++) {
+    if (sum % i === 0) return false;
+  }
+  return sum;
+}
```

Comment:
```suggestion
function solution(nums) {
  let result = 0;

  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        const sum = nums[i] + nums[j] + nums[k];

        if (isPrime(sum)) result++;
      }
    }
  }
  return result;
}

function isPrime(sum) {
  for (let i = 2; i <= Math.sqrt(sum); i++) {
    if (sum % i === 0) return false;
  }
  return sum;
}
```
- 풀이 잘 봤습니다 👍 isPrime에서 반복 횟수를 `i < sum` 를 하는게 아닌 `i <= Math.sqrt(sum)` 를 통해서 반복 횟수를 개선했습니다

Reply:
Math.sqrt(sum)을 해주는 이유가 만약 sum 값이 13 이게 되면 Math.sqrt(13) = 3.605551275463989 이기 때문에,, 결국 i 를 13까지 반복하지 않고 3 까지만 반복해도 같은 값이 도출되기 때문인가요!!! 우와,,,😮